### PR TITLE
[Model] inductive learning for GAT.

### PR DESCRIPTION
**Pull request by sufeidechabei**
## Changes
1. Stable the softmax function in GAT.
2. Add ppi dataset for GAT's training.

### CODE REVIEW

**Comment by jermainewang** [comment]
@sufeidechabei due to the merge of #390 , there are some conflicts. Would you adapt to the latest GAT codes in `gat.py`? The rest trainer scripts could be reused.

File: examples/pytorch/gat/train_cite.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(description='GAT')
+    register_data_args(parser)
+    parser.add_argument("--gpu", type=int, default=1,
+                        help="which GPU to use. Set -1 to use CPU.")
+    parser.add_argument("--epochs", type=int, default=300,
+                        help="number of training epochs")
+    parser.add_argument("--heads", type=list, default=[4, 4, 1],
```
Comment (jermainewang) [inline examples/pytorch/gat/train_cite.py]: Default is [8, 1].

File: python/dgl/data/ppi.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import numpy as np
+from .utils import download, extract_archive, get_download_dir, _get_dgl_url
+from sklearn.preprocessing import StandardScaler
+import networkx as nx
+from networkx.readwrite import json_graph
+import json
+
+_url = 'dataset/ppi.zip'
+
+class ppi_dataset(object):
```
Comment (jermainewang) [inline python/dgl/data/ppi.py]: `PPIDataset`

Comment (jermainewang) [reply python/dgl/data/ppi.py]: Also more docstring. Explain what is each sample looks like.

Comment (jermainewang) [reply python/dgl/data/ppi.py]: Also, try to make it compatible with torch's Dataset interface. Usually, you need to at least implement `__getitem__` and `__len__`. You could have a look at the [MiniGCDataset](https://docs.dgl.ai/api/python/data.html#mini-graph-classification-dataset).
